pkg/pagerduty/pagerduty.go

```diff
@@ -126,6 +128,50 @@ func (p PagerDuty) AddNote(incidentID string, noteContent string) error {
 	return nil
 }
 
+// ExtractExternalIDFromCGHMAlertBody extracts from an Alert body until an external ID
+// the function is a member function but doesn't use any of the types
+func (_ PagerDuty) ExtractExternalIDFromCGHMAlertBody(data map[string]interface{}) (string, error) {
+
+	var err error
+	externalBody := externalCHGMAlertBody{}
+
+	err = mapstructure.Decode(data, &externalBody)
+	if err != nil {
+		return "", fmt.Errorf("cannot marshal externalCHGMAlertBody: %w", err)
+	}
+
+	internalRawData := externalBody.Details.Notes
+	// check if the is a notes field, the default struct is provided if no match is found
+	if internalRawData == "" {
+		return "", AlertBodyDoesNotHaveNotesFieldErr{}
+	}
+
+	internalBody := internalCHGMAlertBody{}
+	err = yaml.Unmarshal([]byte(internalRawData), &internalBody)
+	if err != nil {
+		return "", NotesParseErr{Err: err}
+	}
+
+	// TODO: verify if no ClusterID is missing (doesn't break the test case, but should be noted
+
+	return internalBody.ClusterID, nil
+}
+
+// externalCHGMAlertBody is a struct for manipulating CHGM body until the note object
+type externalCHGMAlertBody struct {
+	// Details are the alert details
+	Details struct {
+		// Notes are the user given notes on the cluster
+		Notes string `mapstructure:"notes"`
```

Comment:
this type is exported by the external lib?

Reply:
yeah, the `mapstructure` helps the external lib comprehend what is going on inside the struct